### PR TITLE
Add an alarm clock plugin

### PR DESCRIPTION
**Pull request by YosiWeinreich**

### CODE REVIEW

**Comment by yosi654** [comment]
Add an alarm clock plugin

File: Plugins/AlarmClock/AlarmClockPlugin.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+namespace AlarmClock
+{
+    public class AlarmClockPlugin : IPluginWithScheduler
+    {
+        IScheduler _scheduler;
+        public AlarmClockPlugin(IScheduler scheduler) => _scheduler = scheduler;
+
+        public static string _Id = "Alarm-Clock";
+        public string Id => _Id;
+        public PluginOutput Execute(PluginInput input)
```
Comment (YacovGold) [inline Plugins/AlarmClock/AlarmClockPlugin.cs:19]: Please add enter between functions/vars

File: Plugins/AlarmClock/AlarmClockPlugin.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        {
+            DateTime dateTime = DateTime.Now;
+            var sh = input.Message.Split(':').First();
+            var sm = input.Message.Split(':').Skip(1).First();
+
+            if (double.TryParse(sh, out double a) && double.TryParse(sm, out double b))
+            {
+
```
Comment (YacovGold) [inline Plugins/AlarmClock/AlarmClockPlugin.cs]: Extra enter

File: Plugins/AlarmClock/AlarmClockPlugin.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        public static string _Id = "Alarm-Clock";
+        public string Id => _Id;
+        public PluginOutput Execute(PluginInput input)
+        {
+            DateTime dateTime = DateTime.Now;
+            var sh = input.Message.Split(':').First();
+            var sm = input.Message.Split(':').Skip(1).First();
+
+            if (double.TryParse(sh, out double a) && double.TryParse(sm, out double b))
```
Comment (YacovGold) [inline Plugins/AlarmClock/AlarmClockPlugin.cs]: 1. You want double or int?
2. Do you use the output afterword's? - if so, save it

File: Plugins/AlarmClock/AlarmClockPlugin.cs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            {
+
+                int h = int.Parse(sh);
+                int m = int.Parse(sm);
+                var ts = new TimeSpan(h, m, 0);
+                var tsNew = new TimeSpan(dateTime.Hour, dateTime.Minute, 0);
+                var tsNew2 = ts - tsNew;
+
```
Comment (YacovGold) [inline Plugins/AlarmClock/AlarmClockPlugin.cs:31]: Extra enters
One enter is enough here

File: Infrastructure/PluginsManager.cs [hunk 5]
```diff
@@ -49,6 +50,10 @@ public IPlugin CreatePlugin(string id)
             {
                 return new CountWordPlugin();
             }
+            else if (id == AlarmClockPlugin._Id)
+            {
+                return new AlarmClockPlugin(new Scheduler(this));
+            }
             else
```
Comment (YacovGold) [inline Infrastructure/PluginsManager.cs]: Extra enter

**Pull request merged**